frame/support/src/weights.rs

```diff
@@ -530,6 +533,90 @@ impl RuntimeDbWeight {
 	}
 }
 
+/// One coefficient and its position in the `WeightToFeePolynomial`.
+///
+/// One term of polynomial is calculated as:
+///
+/// ```ignore
+/// coeff_integer * x^(degree) + coeff_frac * x^(degree)
+/// ```
+///
+/// The `negative` value encodes whether the term is added or substracted from the
+/// overall polynomial result.
+#[derive(Clone, Encode, Decode)]
+pub struct WeightToFeeCoefficient<Balance> {
+	/// The integral part of the coefficient.
+	pub coeff_integer: Balance,
+	/// The fractional part of the coefficient.
+	pub coeff_frac: Perbill,
+	/// True iff the coefficient should be interpreted as negative.
+	pub negative: bool,
+	/// Degree/exponent of the term.
+	pub degree: u8,
+}
+
+/// A list of coefficients that represent one polynomial.
+pub type WeightToFeeCoefficients<T> = SmallVec<[WeightToFeeCoefficient<T>; 4]>;
```

Comment:
Why have the small vec? We only ever use a polynomial with a degree of 1...

Reply:
If we replace that by an array with say length one every runtime implementer needs to change that type alias if they want to use a polynomial with more than one term. That means they need to fork substrate rather than just specify it as dependency.

Reply:
Sure. Question is, given that we only need a linear function, and given that the code to support arbitrary polynomials is non-trivial, why implement it? Is there some strong expectation that other teams will need it (and also that we should dedicate effort to supporting it, rather than merely allowing it)?

Reply:
There is no strong expectation of other teams needing it, although I could imagine something like a fast payments chain with degree > 1 to make heavy transactions very expensive and create blocks with large numbers of lightweight transactions (although the validators of course would very much prioritize the heavy ones).

I'm fine with a linear function; my initial reasoning was that an abstract to linear change was a large restriction, while you can still express a lot in a polynomial. But if long term we plan to move to something like expressing abstract fee calculation as a constant function within the metadata then we can keep this simple and just make it linear only.

Reply:
I fee like the "wasm blob in metadata" is to far away for not even allowing chains to add a static amount of Balance in their calculation for the time being.